include/quicr/fetch_track_handler.h

```diff
@@ -0,0 +1,164 @@
+         * @brief Receive status codes
+         */
+        enum class Error : uint8_t
+        {
+            kOk = 0,
+            kNotAuthorized,
+            kObjectNotAvailable,
+        };
+
+        /**
+         * @brief  Status codes for the track
+         */
+        enum class Status : uint8_t
+        {
+            kOk = 0,
+            kNotConnected,
+            kFetchError,
+            kNotAuthorized,
+            kObjectNotAvailable,
+            kPendingFetchResponse,
+        };
+
+      protected:
+        /**
+         * @brief Fetch track handler constructor
+         *
+         * @param full_track_name Full track name struct.
+         * @param priority The priority of the track.
+         * @param group_order The group order to use.
+         */
+        FetchTrackHandler(const FullTrackName& full_track_name,
+                          messages::ObjectPriority priority,
+                          messages::GroupOrder group_order,
+                          messages::GroupId start_group,
+                          messages::GroupId end_group)
+          : BaseTrackHandler(full_track_name)
+          , priority_(priority)
+          , group_order_(group_order)
+          , start_group_(start_group)
+          , end_group_(end_group)
+        {
+        }
+
+      public:
+        /**
+         * @brief Create shared Fetch track handler
+         *
+         * @param full_track_name Full track name struct.
+         * @param priority The priority of the track.
+         * @param group_order The group order to use.
+         */
+        static std::shared_ptr<FetchTrackHandler> Create(const FullTrackName& full_track_name,
+                                                         messages::ObjectPriority priority,
+                                                         messages::GroupOrder group_order,
+                                                         messages::GroupId start_group,
+                                                         messages::GroupId end_group)
+        {
+            return std::shared_ptr<FetchTrackHandler>(
+              new FetchTrackHandler(full_track_name, priority, group_order, start_group, end_group));
+        }
+
+        /**
+         * @brief Get the status
+         *
+         * @returns The status of the fetch
+         */
+        constexpr Status GetStatus() const noexcept { return status_; }
+
+        /**
+         * @brief Notification of received [full] data object
+         *
+         * @details Event notification to provide the caller the received full data object
+         *
+         * @warning This data will be invalided after return of this method
+         *
+         * @param object_headers    Object headers, must include group and object Ids
+         * @param data              Object payload data received, **MUST** match ObjectHeaders::payload_length.
+         */
+        virtual void ObjectReceived([[maybe_unused]] const ObjectHeaders& object_headers,
```

Comment:
We need to figure out a way to detect if one of `ObjectReceived` or `PartialObjectReceived` is implemented. If one is implemented, then it's a valid fetch handler, if neither are implemented, it's an invalid handler and we shouldn't allow it.